wbooks/src/redux/book/reducer.ts

```diff
@@ -3,12 +3,24 @@ import { BookState } from '@interfaces/Book';
 
 import { actions } from './actions';
 
+interface Action {
+  type: string;
+  target: string;
+  payload: string;
+}
+
 const initialState: BookState = completeState({
-  description: { books: [], filteredBooks: { books: [], query: '' } }
+  description: { books: [], filterQuery: '' }
 });
 
 const bookReducer = {
-  primaryActions: [actions.GET_BOOKS, actions.GET_FILTERED_BOOKS]
+  primaryActions: [actions.GET_BOOKS],
+  override: {
+    [actions.SET_FILTER_QUERY]: (state: BookState, action: Action) => ({
+      ...state,
+      [action.target]: action.payload
+    })
+  }
```

Comment:
Check this https://github.com/Wolox/redux-recompose/tree/master/src/effects/onReadValue

Reply:
o.O Done!